tiledb/api/src/array/domain/strategy.rs

```diff
@@ -67,10 +78,40 @@ mod tests {
     fn domain_eq_reflexivity() {
         let ctx = Context::new().expect("Error creating context");
 
-        proptest!(|(maybe_domain in prop_domain(Default::default()))| {
-            let domain = maybe_domain.create(&ctx)
+        proptest!(|(domain in any::<DomainData>())| {
+            assert_eq!(domain, domain);
+            assert!(domain.option_subset(&domain));
+
+            let domain = domain.create(&ctx)
                 .expect("Error constructing arbitrary domain");
             assert_eq!(domain, domain);
         });
     }
+
+    #[test]
+    fn domain_shrinking() {
+        let strat = any::<DomainData>();
+
+        let mut runner =
+            proptest::test_runner::TestRunner::new(Default::default());
+
+        let mut value = loop {
+            let value = strat.new_tree(&mut runner).unwrap();
+            if value.current().dimension.len() > 4 {
+                break value;
+            }
+        };
+
+        let init = value.current();
+        for _ in 0..1024 {
```

Comment:
Is 1024 here just "a big number" so that we're guaranteed to reduce to the minimum number of dimensions? Could we just use `init.dimension.len()` or is it not guaranteed that simplify removes a dimension?

Reply:
Pretty much yep.  I'm not sure every simplify step will remove a dimension, so I picked a number.  But it would be better to read a value from `proptest::test_runner::Config::default()` since that is more representative of what our tests will actually do.